python/triton/tools/compile.py

```diff
@@ -106,9 +106,10 @@ def constexpr(s):
     # compile ast into cubin
     for h in hints.values():
         assert h in [1, 16], f"Only 1 and 16 are valid hints, got {h}"
-    divisible_by_16 = [i for i, h in hints.items() if h == 16]
-    equal_to_1 = [i for i, h in hints.items() if h == 1]
-    attrs = triton.compiler.AttrsDescriptor(divisible_by_16=divisible_by_16, equal_to_1=equal_to_1)
+    divisible_by_16 = [(i, 16) for i, h in hints.items() if h == 16]
+    equal_to_1 = [(i, 1) for i, h in hints.items() if h == 1]
```

Comment:
Why do we need to pass the extra `16` and `1` here?

Reply:
Those would be the values of the specializations. Basically the value that the property of the attribute will be set to. The point is that the class is trying to do two things
1. It is taking params and arguments, and it is coming up with something like `prop1: ( (p0,val), (p3,val), (p4,val))`
2. It is combing those params together to form a list of param properties `p0 : {prop1:val, prop2:val,} ....
This is why I am passing those values around